src/BoundaryConditions/flux_calculations.jl

```diff
@@ -1,61 +1,130 @@
 """
-    get_boundary_flux(model, ::NoFluxCondition, var::Fields.Field, Y, Ya)
+    get_boundary_flux(model, ::NoFlux, var::Fields.Field, Y, Ya)
 
-    Vertical flux assignment for no-flux conditions (i.e. free-slip wall)
+    Vertical flux assignment for no-flux conditions (i.e., insulating wall)
 """
-function get_boundary_flux(model, ::NoFluxCondition, var::Fields.Field, Y, Ya)
+function get_boundary_flux(model, ::NoFlux, var::Fields.Field, Y, Ya)
     FT = eltype(Y)
-    flux = Geometry.WVector(FT(0))
+    flux = Geometry.Contravariant3Vector(FT(0))
 end
 
 """
-    get_boundary_flux(model, bc::DragLawCondition, uv, Y, Ya)
+    get_boundary_flux(model, ::NoVectorFlux, var::Fields.Field, Y, Ya)
 
-    Vertical momentum fluxes for a bulk-formulation drag law, given momentum exchange coefficients
+    Vertical flux assignment for an AxisTensor with CovariantAxis{(1,2)} in no-flux conditions (i.e., insulating wall)
 """
-function get_boundary_flux(model, bc::DragLawCondition, uv, Y, Ya)
+function get_boundary_flux(model, ::NoVectorFlux, var::Fields.Field, Y, Ya)
     FT = eltype(Y)
-    coefficients =
-        eltype(bc.coefficients) == FT ? bc.coefficients : bc.coefficients(Y, Ya)
+    flux =
+        Geometry.Covariant3Vector(FT(0)) ⊗
+        Geometry.Covariant12Vector(FT(0), FT(0))
+end
 
-    uv_1 = first_interior(uv)
+"""
+    get_boundary_flux(model, bc::DragLaw, uv, Y, Ya)
+
+    Vertical momentum fluxes for a bulk-formulation drag law, given the momentum exchange coefficient
+"""
+function get_boundary_flux(model, bc::DragLaw, uv, Y, Ya)
```

Comment:
I think that it would be better to use SF.jl for. compute any flux, if something is missing there we should add it there. The current DragLaw is given by SF.jl:

https://github.com/CliMA/SurfaceFluxes.jl/blob/035a833f59a90d889eadef837491da5b8fa9908e/src/SurfaceFluxes.jl#L589